task/rsync/0.1/rsync.yaml

```diff
@@ -0,0 +1,70 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: rsync
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: rsync, files sync
+    tekton.dev/displayName: "rsync"
+spec:
+  description: >-
+        This task can be used to synchronize local and remote files.
+  workspaces:
+    - name: source
+      description: The directory you want to sync.
+    - name: ssh-directory
+      optional: true
+      description: A .ssh directory with private key, known_hosts, config. It will overwrite the authentication by username and password.
+  params:
+    - name: direction
+      type: string
+      default: "local-to-remote"
+    - name: local-path
+      type: string
+      default: "./"
+    - name: remote-ip
+      type: string
+      default: "0.0.0.0"
+    - name: remote-username
+      type: string
+      default: "root"
+    - name: remote-password
+      type: string
+      default: "password"
+    - name: remote-path
+      type: string
+      default: "~/"
+  steps:
+    - name: rsync
+      image: docker.io/shaowenchen/rsync-sshpass:v1
```

Comment:
can we tag the image with digest as well?

Reply:
fixed.